publish/npm/install.js

```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/env node
+
+const fs = require("fs");
+const path = require("path");
+const https = require("https");
+const { execSync } = require("child_process");
+
+// GitHub release info
+const BINARY_NAME = "pmg";
+
+// Platform and architecture mapping
+function getPlatformInfo() {
+  const platform = process.platform;
+  const arch = process.arch;
+
+  let platformName,
+    archName,
+    extension,
+    isZip = false;
+
+  // Map Node.js platform to release naming (matching GoReleaser output)
+  switch (platform) {
+    case "darwin":
+      platformName = "Darwin_all";
+      extension = ".tar.gz";
+      break;
+    case "linux":
+      platformName = "Linux";
+      extension = ".tar.gz";
+      break;
+    case "win32":
+      platformName = "Windows";
+      extension = ".zip";
+      isZip = true;
+      break;
+    default:
+      throw new Error(`Unsupported platform: ${platform}`);
+  }
+
+  // Map Node.js arch to release naming
+  switch (arch) {
+    case "x64":
+      archName = "x86_64";
+      break;
+    case "arm64":
+      archName = "arm64";
+      break;
+    case "ia32":
+      archName = "i386";
+      break;
+    default:
+      throw new Error(`Unsupported architecture: ${arch}`);
+  }
+
+  // Special case for Darwin - it's all architectures in one
+  if (platform === "darwin") {
+    return {
+      filename: `pmg_${platformName}${extension}`,
```

Comment:
Can we avoid hardcoding `pmg` here and may be used a constant from some `config.js`? That way, all project specific keywords are in a single file and we can easily reuse this `npm` wrapper across our projects